Issue #2169: Update HTML documentation to describe all options for all checks

MEZk opened the pull request.

romani commented:
Please share new website on github.io , this will speadup merge process

mkordas commented on src/xdocs/config_blocks.xml:
```diff
@@ -50,31 +50,36 @@
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_WHILE">LITERAL_WHILE</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">LITERAL_TRY</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FINALLY">LITERAL_FINALLY</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">LITERAL_DO</a>,
```
`LITERAL_DO` is here on 4th position, but in previous column it's on first. Shouldn't it be unified?

MEZk replied:
@mkordas 
Corrected.

mkordas commented:
@MEZk, changes are really impressive. Did you have any tool to generate them? Ideally we should have some tests to confirm that documentation is up-to-date.

romani commented:
@MEZk , it is not clear why not all Checks segments are changed. I expect update to all Checks even that do not have any option right now at xdoc.

MEZk commented:
@romani
For other checks acceptable token sets are equal to required token sets . User has no possibility to configure tokes for a check

MEZk commented:
@mkordas 
No, it was done manually) But UT which checks that all Checks are present in xdoc will be implemented in accordance with https://github.com/checkstyle/checkstyle/issues/819

MEZk commented:
@romani 
http://mezk.github.io/site/index.html

mkordas commented:
@MEZk 
Can you review carefully all the below checks? I noticed in all of them some kind of problems (eg.missing commas, not saying this is subset, not having two first columns etc.)
http://mezk.github.io/site/config_naming.html#LocalFinalVariableName
http://mezk.github.io/site/config_naming.html#LocalVariableName
http://mezk.github.io/site/config_coding.html#InnerAssignment
http://mezk.github.io/site/config_coding.html#MagicNumber
http://mezk.github.io/site/config_javadoc.html#AtclauseOrder
http://mezk.github.io/site/config_whitespace.html#NoWhitespaceAfter
http://mezk.github.io/site/config_whitespace.html#TypecastParenPad
http://mezk.github.io/site/config_coding.html#IllegalType

MEZk commented:
@mkordas 
>http://mezk.github.io/site/config_naming.html#LocalFinalVariableName
http://mezk.github.io/site/config_naming.html#LocalVariableName
http://mezk.github.io/site/config_coding.html#InnerAssignment
http://mezk.github.io/site/config_coding.html#MagicNumber
http://mezk.github.io/site/config_javadoc.html#AtclauseOrder
http://mezk.github.io/site/config_whitespace.html#NoWhitespaceAfter
http://mezk.github.io/site/config_coding.html#IllegalType

Corrected.

http://mezk.github.io/site/config_whitespace.html#TypecastParenPad
Removed 'token' option because acceptable tokens set is equal to required tokens set. User can not configure tokens set for the check.

romani commented:
merged as FF
